packages/api/src/routers/svc/speech.ts

```diff
@@ -21,68 +21,57 @@ export function speechServiceRouter() {
       // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
       body: req.body,
     })
+    const { userId, speechId } = req.body as {
```

Comment:
Is there a way we can do the token check here before setting failure state? Otherwise I think people could set existing speeches to failed that they don't own.

Reply:
My mistake here. I will validate token first